components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/client/ApplicationManagementServiceClient.java

```diff
@@ -119,9 +119,22 @@ public ServiceProvider getApplication(String applicationName) throws AxisFault {
      * @return
      * @throws AxisFault
      */
-    public ApplicationBasicInfo[] getAllApplicationBasicInfo() throws Exception {
+    public ApplicationBasicInfo[] getAllApplicationBasicInfo() throws AxisFault {
+
+        return getApplicationBasicInfo("*");
+    }
+
+    /**
+     * Get all basic application information for a matching filter.
+     *
+     * @param filter Application name filter
+     * @return Application Basic Information array
+     * @throws AxisFault
+     */
+    public ApplicationBasicInfo[] getApplicationBasicInfo(String filter) throws AxisFault {
+
         try {
-            return stub.getAllApplicationBasicInfo();
+            return stub.getApplicationBasicInfo(filter);
         } catch (RemoteException | IdentityApplicationManagementServiceIdentityApplicationManagementException e) {
```

Comment:
Can't we handle in a separate catch block?

Reply:
Since it is the same action in both cases it should be okay to use the same catch block.